lib/Ed25519KeyPair.js

```diff
@@ -11,32 +11,73 @@ const LDKeyPair = require('./LDKeyPair');
 
 class Ed25519KeyPair extends LDKeyPair {
   /**
-   * @param options {object} See LDKeyPair constructor docstring for full list
-   *
-   * @param [options.publicKeyBase58] {string}
-   * @param [options.privateKeyBase58] {string}
+   * @classdesc  An implementation of
+   * [Ed25519 Signature 2018]{@link https://w3c-dvcg.github.io/lds-ed25519-2018/}
+   * for
+   * [jsonld-signatures.]{@link https://github.com/digitalbazaar/jsonld-signatures}
+   * @description Creates an [Ed25519]{@link https://ed25519.cr.yp.to/}
+   * Key Pair usable by a json-ld signature.
+   * @example
+   * > const ops = {
+   *   publicKeyBase58: '2S2Q6MkaFJewa',
+   *   privateKeyBase58: '8JHajxJ1pKnSxwziUGFnuzvB96psLvqejZtgcA7QKoZ2'
+   * };
+   * > const EDKey = new Ed25519KeyPair(ops);
+   * @param {KeyPairOptions} options - Keys must be in Base58
+   * other options most follow
+   * [KeyPairOptions]{@link ./index.md#KeyPairOptions}.
+   * @param {string} options.publicKeyBase58 - Public Key exposed in Signatures.
+   * @param {string} options.privateKeyBase58 - Private Key for signing.
    */
   constructor(options = {}) {
     super(options);
     this.type = 'Ed25519VerificationKey2018';
     this.privateKeyBase58 = options.privateKeyBase58;
     this.publicKeyBase58 = options.publicKeyBase58;
   }
-
+  /**
+   * @description Returns the public key.
+   * @example
+   * > edKeyPair.publicKey;
+   * 2S2Q6MkaFJewa
+   * @implements {LDKeyPair#publicKey}
+   * @readonly
+   * @returns {string} Returns the public key.
+   * @see [publicKey]{@link ./LDKeyPair.md#publicKey}
+   */
   get publicKey() {
     return this.publicKeyBase58;
   }
-
+  /**
+   * @description Returns the private key.
+   * @example
+   * > edKeyPair.privateKey;
+   * 8JHajxJ1pKnSxwziUGFnuzvB96psLvqejZtgcA7QKoZ2
+   * @implements {LDKeyPair#privateKey}
+   * @readonly
+   * @returns {string} Returns the private key.
+   * @see [privateKey]{@link ./LDKeyPair.md#privateKey}
+   */
   get privateKey() {
     return this.privateKeyBase58;
   }
 
   /**
-   * @param [options] {object} See LDKeyPair docstring for full list
-   * @param [options.seed] {(Uint8Array|Buffer)} a 32-byte seed for a
-   *   deterministic key.
+   * @description Static method that generates
+   * a KeyPair with a deterministic seed.
+   * @example
+   * > const ops = {
+   *   seed: new Int8Array(58)
+   * };
+   * > const keyPair = await Ed25519KeyPair.generate(ops);
+   * > keyPair
+   * Ed25519KeyPair { ...
+   * @param {KeyPairOptions} [options={}] - See LDKeyPair
```

Comment:
right now no. I think we can continue looking at other projects because it is just markdown at the end of the say so the template could be modified, however do not quite understand the templates for jsdoc-to-markdown yet on as advanced a level as other projects do. We might want to make this a feature of some sort. I can totally research this and it is obviously doable, but well depends on how much time we want to spend on docs.

Reply:
@dlongley @mattcollier @dmitrizagidulin  ok we do need to discuss this. How are we handling @type stuff elsewhere in the project? Has anyone seen this implemented elsewhere?